src/components/ManageAccountsDrawer/components/ImportAccounts.tsx

```diff
@@ -122,41 +127,13 @@ export const ImportAccounts = ({ chainId, onClose }: ImportAccountsProps) => {
   // But also need to remove accounts that were disabled.
 
   // initial fetch to detect the number of accounts based on the "first empty account" heuristic
-  useEffect(() => {
-    ;(async () => {
-      if (!wallet) return
-
-      let accountNumber = 0
-      const accounts = []
-
-      while (true) {
-        try {
-          const accountResult = await queryClient.fetchQuery(
-            reactQueries.accountManagement.accountIdWithActivity(accountNumber, chainId, wallet),
-          )
-
-          if (!accountResult) break
-
-          const { accountId, hasActivity } = accountResult
-
-          // Add the account before checking if it has activity so user has ability to toggle it if needed.
-          accounts.push({ accountNumber, accountId })
-
-          // If the account has no activity, it's the first empty account.
-          if (!hasActivity) {
-            break
-          }
-        } catch (error) {
-          console.error(error)
-          break
-        }
-
-        accountNumber++
-      }
+  const { data: allAccountIdsWithActivity } = useQuery(
+    accountManagement.allAccountIdsWithActivity(chainId, wallet, NUM_ADDITIONAL_EMPTY_ACCOUNTS),
+  )
```

Comment:
@gomesalexandre , looking forward to some recommendations on staletime etc as these will vary over time as users use their wallets

Reply:
IMO since we have `wallet` as a dep meaning this is nice and safe for a specific wallet, we should be able to keep things stale and garbage-collected with very long and conservative times i.e for the lifetime of a tab.

Even if we plan to reuse this and some places would need more freshness, each consumer can set their own options to get fresh-er data, which `<ImportAccounts />` probably doesn't need during the lifetime of a tab.

We can always `invalidateQueries` programmatically in e.g `<TransactionsProvider />` if needed